### PR TITLE
Clean up and align NVMe driver with tianocore as much as possible.

### PR DESCRIPTION
**Pull request by spbrogan**

### CODE REVIEW

**Comment by spbrogan** [comment]
@corthon  - what do you think about removing all the debugging comments.  I don't think they are worth carrying mu_changes

File: MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressHci.c [hunk 1]
```diff
@@ -1126,9 +1126,7 @@ NvmeUnregisterShutdownNotification (
 {
   EFI_STATUS                      Status;
   EFI_RESET_NOTIFICATION_PROTOCOL *ResetNotify;
-  //MS_CHANGE - BEGIN
-  ReportStatusCode((EFI_ERROR_MAJOR | EFI_ERROR_CODE), (EFI_IO_BUS_SCSI | EFI_IOB_EC_INTERFACE_ERROR));
```
Comment (matthewfcarlson) [inline MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressHci.c:1130]: Would these report status code removals be good candidates for instead pivoting to emitting telemetry here?

**Review by matthewfcarlson — APPROVED** [review]

**Comment by matthewfcarlson** [comment]
I agree- I think this debugging was probably helpful when developing/first implementing IoMMU, but not worth the MU_CHANGE and accompanying commit. When we need to debug issues, we'll probably add our own print statements that relevant to the issue at hand.

File: MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressHci.c (hunk 1 above)
Comment (spbrogan) [reply MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressHci.c:1130]: no.  telemetry is not in basecore yet

**Pull request closed without merging**